### ISSUE TITLE
Have not found Monte Carlo Sampling in the code

### ISSUE THREAD

**Issue by fangkuann** [google/active-qa#21]
Hi, 
 Thanks for releasing the code for active-qa. 
  After browsing the code, I did not find Monte-Carlo Sampling in the training stage. It seems that each training instance consists of only one  「query, reformulated_query, reward」 tuple.  Therefore, the reward is the same for each token in one reformulated query. 
I don't know whether the suspicion is right. If it is right,  what will model perform with or without Monte-Carlo sampling? Maybe using only one instance for Monte Carlo sampling is like the relation between stochastic gradient descent and gradient descent?
Thank you

**Comment by godfly** [comment]
me the same, have you finger out the problem?or get a new version code? @fangkuann

**Comment by fangkuann** [comment]
I didn't try that. But we apply the paper's training method in our query rewrite module, then the retrieve performance could be enhanced. We further improved this method by adding a value network, for details you may refer to this Chinese technic article https://www.6aiq.com/article/1577969687897.

**Comment by fangkuann** [comment]
> me the same, have you finger out the problem?or get a new version code? @fangkuann

I didn't try that. But we apply the paper's training method in our query rewrite module, then the retrieve performance could be enhanced. We further improved this method by adding a value network, for details you may refer to this Chinese technic article https://www.6aiq.com/article/1577969687897.

**Comment by godfly** [comment]
> > me the same, have you finger out the problem?or get a new version code? @fangkuann
> 
> I didn't try that. But we apply the paper's training method in our query rewrite module, then the retrieve performance could be enhanced. We further improved this method by adding a value network, for details you may refer to this Chinese technic article https://www.6aiq.com/article/1577969687897.

 @fangkuann  Yes, I follow this article to here. May I ask some question by email? I couldn't found a way to concat you. Send a message to yanggodfly1994@gmail.com if it's ok, thanks a lot
